src/Renci.SshNet/SshCommand.cs

```diff
@@ -294,6 +294,9 @@ public string EndExecute(IAsyncResult asyncResult)
                             _channel.Close();
                         }
 
+                        this._session.Disconnected -= Session_Disconnected;
+                        this._session.ErrorOccured -= Session_ErrorOccured;

```

Comment:
I don't think we should unregister from these events here, as the same SshCommand can be re-executed after its finished.